<?xml version="1.0" ?>
<launch>

    <arg name="tabletop_segmentation_points_in" default="camera/depth_registered/points" />
    <arg name="tabletop_segmentation_convert_to_base_link" default="true" />
    <arg name="tabletop_segmentation_clustering_voxel_size" default="0.002" />
    <arg name="flatten_table" default="false"/>
    <arg name="processing_frame" default="base_link"/>
    <arg name="up_direction" default="1.0"/>
    <arg name="z_filter_min" default="-0.1"/>
    <arg name="z_filter_max" default="0.5"/>
    <arg name="y_filter_min" default="-1.0"/>
    <arg name="y_filter_max" default="1.2"/>
    <arg name="x_filter_min" default="0.2"/>
    <arg name="x_filter_max" default="1.0"/>
    <arg name="table_z_filter_min" default="-0.5"/>
    <arg name="table_z_filter_max" default="-0.01"/>

    <node pkg="mantis_perception" name="tabletop_segmentation" type="mantis_segmentation" respawn="true" output="screen">
	<!--topic remapping-->
        <remap from="cloud_in" to="$(arg tabletop_segmentation_points_in)" />
        <remap from="segmentation_srv" to="tabletop_segmentation" />
        <remap from="markers_out" to="tabletop_segmentation_markers" />

	<param name="clustering_voxel_size" value="$(arg tabletop_segmentation_clustering_voxel_size)" />
	<param name="inlier_threshold" value="300" />
	<param name="plane_detection_voxel_size" value="0.01" />
	<param name="cluster_distance" value="0.02" /><!-- minimum distance between clusters -->
	<param name="min_cluster_size" value="500" />
	<param name="flatten_table" value="$(arg flatten_table)" />

	<!-- processing and filtering frame -->
	<!-- all clouds converted to and processed in base link frame -->
	<param if="$(arg tabletop_segmentation_convert_to_base_link)" name="processing_frame" value="$(arg processing_frame)" />
	<param if="$(arg tabletop_segmentation_convert_to_base_link)" name="up_direction" value="$(arg up_direction)" />
	<param if="$(arg tabletop_segmentation_convert_to_base_link)" name="z_filter_min" value="$(arg z_filter_min)" />
	<param if="$(arg tabletop_segmentation_convert_to_base_link)" name="z_filter_max" value="$(arg z_filter_max)" />
	<param if="$(arg tabletop_segmentation_convert_to_base_link)" name="y_filter_min" value="$(arg y_filter_min)" />
	<param if="$(arg tabletop_segmentation_convert_to_base_link)" name="y_filter_max" value="$(arg y_filter_max)" />
	<param if="$(arg tabletop_segmentation_convert_to_base_link)" name="x_filter_min" value="$(arg x_filter_min)" />
	<param if="$(arg tabletop_segmentation_convert_to_base_link)" name="x_filter_max" value="$(arg x_filter_max)" />
	<param if="$(arg tabletop_segmentation_convert_to_base_link)" name="table_z_filter_min" value="$(arg table_z_filter_min)" />
	<param if="$(arg tabletop_segmentation_convert_to_base_link)" name="table_z_filter_max" value="$(arg table_z_filter_max)" />			

	<!-- all clouds will be processed in the incoming frame, which is -->
	<!-- assumed to the the camera's optical frame -->
	<param unless="$(arg tabletop_segmentation_convert_to_base_link)" name="processing_frame" value="" />
	<param unless="$(arg tabletop_segmentation_convert_to_base_link)" name="up_direction" value="-1.0" />
	<param unless="$(arg tabletop_segmentation_convert_to_base_link)" name="z_filter_min" value="0.4" />
	<param unless="$(arg tabletop_segmentation_convert_to_base_link)" name="z_filter_max" value="1.25" />
	<param unless="$(arg tabletop_segmentation_convert_to_base_link)" name="y_filter_min" value="-1.2" />
	<param unless="$(arg tabletop_segmentation_convert_to_base_link)" name="y_filter_max" value="1.2" />
	<param unless="$(arg tabletop_segmentation_convert_to_base_link)" name="x_filter_min" value="-1.2" />
	<param unless="$(arg tabletop_segmentation_convert_to_base_link)" name="x_filter_max" value="1.2" />
	<param unless="$(arg tabletop_segmentation_convert_to_base_link)" name="table_z_filter_min" value="0.01" />
	<param unless="$(arg tabletop_segmentation_convert_to_base_link)" name="table_z_filter_max" value="0.5" />			

    </node>

</launch>
